reactor-netty-http/src/main/java/reactor/netty/http/server/AbstractHttpServerMetricsHandler.java

```diff
@@ -182,12 +205,22 @@ public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise)
 
 	@Override
 	public void channelRead(ChannelHandlerContext ctx, Object msg) {
+		HttpServerOperations ops = null;
 		try {
 			if (msg instanceof HttpRequest) {
+				reset();
 				ChannelOperations<?, ?> channelOps = ChannelOperations.get(ctx.channel());
 				if (channelOps instanceof HttpServerOperations) {
-					HttpServerOperations ops = (HttpServerOperations) channelOps;
-					startRead(ops, uriTagValue == null ? ops.path : uriTagValue.apply(ops.path), methodTagValue.apply(ops.method().name()));
+					ops = (HttpServerOperations) channelOps;
+					if (!initialized) {
```

Comment:
is it necessary to do this `if (!initialized) {` ?
The reset() method that has been invoked at line 211 has reset the `initialized` field to `false`, so maybe we can remove the `if` and always initialize the fields , or am I missing something else ?

Reply:
yeah true, let me fix it